arangod/Cluster/Maintenance.cpp

```diff
@@ -522,12 +521,19 @@ arangodb::Result arangodb::maintenance::executePlan (
   report.add(VPackValue(ACTIONS));
   { VPackArrayBuilder a(&report);
     // enact all
-    for (auto const& action : actions) {
+    for (auto& action : actions) {
       LOG_TOPIC(DEBUG, Logger::MAINTENANCE)
-        << "adding action " << action << " to feature ";
+        << "phase one adding action " << action << " to feature ";
       { VPackObjectBuilder b(&report);
         action.toVelocyPack(report);
       }
+
+      auto const& name = action.get(NAME);
```

Comment:
spaghetti code.  Everything specific to one type of action needs to happen within that action's class.  See notes on virtualized setState().

Reply:
cannot be avoided for now. cannot pass the feature all the way to the constructor. would change a lot. this is marked for later consideration,